### PR TITLE
add jwt-token validation test for host api

### PR DESCRIPTION
**Pull request by pranavs18**

### CODE REVIEW

**Comment by ibuildthecloud** [comment]
If you run these tests they will fail.  Have you determined why yet?

**Comment by pranavs18** [comment]
So this is what i have figured out @ibuildthecloud  - 
1) The IP returned by doing a host stats() returns 127.0.0.1 as the Host IP and perhaps on a MAC it's not explicitly set to the boot2docker's IP as no one is listening on 127.0.0.1
The host returns 127.0.0.1 in the stats socket and when we try to connect to that, which is my laptop localhost, it have anything listening to it and thus the Websocket connection is throws a 61 Err Connection Refused exception.This is why the test would fail.

2) I tried to pass the boot2docker's IP explicitly in the URL and that resulted in Websocket 401 Handshake error (Failed authentication). Now this probably means the token sent in the request is not getting authenticated by the host-api service. I was trying to look into how can I resolve this handshake error. Few questions rose in my mind then -  
-Does the Go agent code properly handle the token authentication , even if i enable the two parameters in the host-api process launch ? 
- Assuming, the token authentication is handled correctly, then the issue is something to do with token being passed to the host_api?

**Comment by pranavs18** [comment]
Tests have been updated. Works with both boot2docker and GCE .

File: tests/validation/cattlevalidationtest/core/test_host_api.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from common_fixtures import *  # NOQA
+from test_docker import docker_context
+import websocket as ws
+# work around flake8 issue
+docker_context
```
Comment (ibuildthecloud) [inline tests/validation/cattlevalidationtest/core/test_host_api.py]: What is this?

Comment (pranavs18) [inline tests/validation/cattlevalidationtest/core/test_host_api.py]: I just followed the pattern from existing test_host_stats.py file here to use docker_context - https://github.com/rancherio/cattle/blob/master/tests/integration/cattletest/core/test_host_stats.py

**Pull request closed without merging**